ckanext/datastore/writer.py

```diff
@@ -0,0 +1,169 @@
+
+    :param response: file-like or response-like object for writing
+        data and headers (response-like objects only)
+    :param columns: list of column names
+    :param name: file name (for headers, response-like objects only)
+    :param bom: True to include a UTF-8 BOM at the start of the file
+
+    >>> with csv_writer(response, columns) as d:
+    >>>    d.writerow(row1)
+    >>>    d.writerow(row2)
+    '''
+
+    if hasattr(response, u'headers'):
+        response.headers['Content-Type'] = b'text/csv; charset=utf-8'
+        if name:
+            response.headers['Content-disposition'] = (
+                b'attachment; filename="{name}.csv"'.format(
+                    name=encode_rfc2231(name)))
+    wr = unicodecsv.writer(response, encoding=u'utf-8')
+    if bom:
+        response.write(UTF8_BOM)
+    wr.writerow(columns)
+    yield wr
+
+
+@contextmanager
+def tsv_writer(response, columns, name=None, bom=False):
+    u'''Context manager for writing UTF-8 TSV data to response
+
+    :param response: file-like or response-like object for writing
+        data and headers (response-like objects only)
+    :param columns: list of column names
+    :param name: file name (for headers, response-like objects only)
+    :param bom: True to include a UTF-8 BOM at the start of the file
+
+    >>> with tsv_writer(response, columns) as d:
+    >>>    d.writerow(row1)
+    >>>    d.writerow(row2)
+    '''
+
+    if hasattr(response, u'headers'):
+        response.headers['Content-Type'] = (
+            b'text/tab-separated-values; charset=utf-8')
+        if name:
+            response.headers['Content-disposition'] = (
+                b'attachment; filename="{name}.tsv"'.format(
+                    name=encode_rfc2231(name)))
+    wr = unicodecsv.writer(
+        response, encoding=u'utf-8', dialect=unicodecsv.excel_tab)
+    if bom:
+        response.write(UTF8_BOM)
+    wr.writerow(columns)
+    yield wr
+
+
+@contextmanager
+def json_writer(response, columns, name=None, bom=False):
+    u'''Context manager for writing UTF-8 JSON data to response
+
+    :param response: file-like or response-like object for writing
+        data and headers (response-like objects only)
+    :param columns: list of column names
+    :param name: file name (for headers, response-like objects only)
+    :param bom: True to include a UTF-8 BOM at the start of the file
+
+    >>> with json_writer(response, columns) as d:
+    >>>    d.writerow(row1)
+    >>>    d.writerow(row2)
+    '''
+
+    if hasattr(response, u'headers'):
+        response.headers['Content-Type'] = (
+            b'application/json; charset=utf-8')
+        if name:
+            response.headers['Content-disposition'] = (
+                b'attachment; filename="{name}.json"'.format(
+                    name=encode_rfc2231(name)))
+    if bom:
+        response.write(UTF8_BOM)
+    response.write(
+        b'{\n  "columns": %s,\n  "data": [' % json.dumps(
```

Comment:
Sure I can change that but the data here is a list of lists not a list of objects. Should we change `datastore_create` so that it accepts a list of lists too?